IntroToCSharp/Pages/Activities/TheTypeSystem.razor

```diff
@@ -0,0 +1,151 @@
+@inherits CaptainCoder.Page
+@page "/activities/the-type-system"
+
+<MudText Typo="Typo.h4" Style="font-weight:bold">The Type System</MudText>
+
+<Section Title="Introduction">
+    <MudText>As we learned in the <MudLink Href="/activities/variables">Variables</MudLink> lesson, each variables must
+        have a <Keyword Word="type"></Keyword> that describes its <Keyword Word="data"></Keyword>. There are <Keyword
+            Word="14"></Keyword> different types of data that programmers can use. Defining data types for our variables
+        allows the computer to know how to store it while our program runs.
+    </MudText>
+</Section>
+<Section Title="Integers">
+    <MudText>
+        There are 8 different integer types that each use a certain amount of bytes. The amount of bytes in a type
+        determines the amount of memory usage.
+    </MudText>
+    <Note>You do not need to memorize this table but it is good to be aware.</Note>
+    <DataTable Headers="@intHeaders" Rows="@intRows"></DataTable>
+    <CodeBlock>
+        int childAge = 8;
+    </CodeBlock>
+    <Questions Title="Let's Practice!">
+        <Message>
+            Above is an example of declaring an int variable. Seems simple, right? Well, sometimes it can take some time
+            to think of the
+            appropriate name. In the lesson on variables, you learned about variable naming guidlines. Apply your
+            knowledge here
+            while you practice declaring int variables!
+        </Message>
+        <ChildContent>
+            <Question Text="Declare an int variable for your mother's age!" />
+            <Question Text="Declare an int variable for your father's age!" />
+        </ChildContent>
+    </Questions>
+</Section>
+
+<Section Title="Floating-Point Numbers">
+    <MudText>Integers are great for whole numbers, but what happens when we want to store decimals or even repeating
+        decimals like π? That's when programmers use floating-point data types. We will learn about 3 floating-point
+        data types: float, double, decimal.</MudText>
+    <Note>The phrase "floating-point" comes repeating or "floating" digits after the decimal or "point".</Note>
+    <MudElement Class="ma-4">
+        <DataTable Headers=@floatHeaders Rows=@floatRows></DataTable>
+    </MudElement>
+    <MudText>Float and double values are not exactly precise, they are simply estimations of those defined values.
+        But a decimal value is precise. You will see this in the below example.</MudText>
+
+    <CodeBlock ReplIt="JoshuaJoshJosh1/FloatPointExample">
+    using System;
+class FloatingPointNumbers
+{
+	public static void Main()
+	{
```

Comment:
Let's stick to using examples without a `Main` method:

```csharp
		 float floatNumber = 1.5f;
 		 double doubleNumber = 1.5;
   		 decimal decimalNumber = 1.5m;
  		 Console.WriteLine(floatNumber - 1.0f);
   		 Console.WriteLine(doubleNumber - 1);
   		 Console.WriteLine(decimalNumber - 1.0m);
```

Reply:
All of the answers come out to 0.5, so need to fix this example or change it completely